### PR TITLE
Add a deprecation notice and plan

### PR DESCRIPTION
**Pull request by glbrntt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 [![Build Status](https://travis-ci.org/grpc/grpc-swift.svg?branch=master)](https://travis-ci.org/grpc/grpc-swift)
 
+# ‼️ Deprecation Notice ‼️
+
+`v0.x` **will soon be deprecated in favor of a new implementation, please see
```
Comment (MrMage) [inline README.md]: nit: replace `,` with `;`

Also, I would provide a few more info here. Maybe "gRPC Swift versions 0.x based on gRPC-Core will soon be replaced with a re-implementation based on SwiftNIO. We strongly suggest using the [`nio` branch](https://github.com/grpc/grpc-swift/tree/nio) for new projects. Please refer to [DEPRECATION.md](DEPRECATION.md) for more information on the deprecation process."

File: DEPRECATION.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# gRPC Swift 0.x deprecation
+
+## What is this document?
+
+This document outlines the current plan to deprecate the `0.x` release series of gRPC Swift.
+
+## What is happening?
+
+The `0.x` release series of the gRPC Swift project will be deprecated in favour of a new implementation.
```
Comment (MrMage) [inline DEPRECATION.md]: (Maybe) See my comment in README.md to provide more details in this first paragraph.

File: DEPRECATION.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+In the coming weeks this branch (currently `master`) containing the `0.x` releases will be renamed `cgrpc`. The `nio` branch containing the [new implementation][nio-branch] will subsequently be renamed to `master` and become the default branch. A `1.0.0` tag will also be created.
+
+Version | Branch Now | Branch After Deprecation
+--------|------------|-------------------------
+`0.x`   | `master`   | `cgrpc`
+`1.x`   | `nio`      | `master`
+
+## What is it being replaced with?
+
+We have rewritten gRPC Swift on top of [SwiftNIO][nio], an open-source asynchronous event-driven networking framework created by Apple. Our implementation will be written in Swift and will not wrap the gRPC core C library.
```
Comment (MrMage) [inline DEPRECATION.md]: nit: I think Core is capitalized.

File: DEPRECATION.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+## Why is this happening?
+
+There are a number of reasons we have rewritten gRPC Swift:
+
+- The `0.x` releases are built on top of a C interface to the gRPC core library provided by the [gRPC project][grpc]. This led to a number of memory safety issues and is easy to hold incorrectly.
+- Network connectivity changes (e.g. LTE to WiFi, 3G to LTE, etc.) are not handled well by the networking stack in the core library. [Network.framework][network-framework] (where available, see below) alleviates this problem and has integration with [SwiftNIO][nio].
+- [SwiftNIO][nio] has gained a lot of traction in the Swift on Server community due to its performance. We believe we can leverage this to improve the performance of gRPC Swift.
```
Comment (MrMage) [inline DEPRECATION.md]: "performance and overall stability"?

File: DEPRECATION.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+- Vendoring copies of the gRPC core library and BoringSSL is a source of frustration for developers and users.
+
+## What will happen to the `0.x` releases?
+
+We will continue to patch the `cgrpc` branch for security fixes and serious bugs only. There will be no feature development on the `cgrpc` branch and the version of the gRPC core library will not be updated (unless necessary for a security fixes).
```
Comment (MrMage) [inline DEPRECATION.md]: nit: s/fixes/fix/ at the end of the line

File: DEPRECATION.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+## When is this happening?
+
+We plan to deprecate `0.x` and tag `1.0.0` by the end of the year.
```
Comment (MrMage) [inline DEPRECATION.md]: "deprecate versions `0.x`"?
"by the end of 2019"?

File: DEPRECATION.md [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+We have the same [platform support as SwiftNIO][nio-platforms]. That is:
+
+* macOS 10.12+, iOS 10+
+* macOS 10.14+, iOS 12+, or tvOS 12+ (with [Network.framework][network-framework] support via [NIO Transport Services][nio-ts])
+* Ubuntu 14.04+
+
+## Which package managers will be supported for `1.x`?
+
+We will provide support for Swift Package Manager as we believe that its integration with Xcode is the most convient way to manage packages.
```
Comment (MrMage) [inline DEPRECATION.md]: typo: convient

File: DEPRECATION.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+
+Carthage will _not_ be supported as it has been the source of a [number][carthage0] [of][carthage1] [issues][carthage2] [in][carthage3] [the][carthage4] [past][carthage5].
```
Comment (MrMage) [inline DEPRECATION.md]: Also mention that CocoaPods support is not planned for now?

File: DEPRECATION.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+## Can I try this out now?
+
+Absolutely! Head over to the `nio` [branch][nio-branch] and check out the [quick-start guide][quick-start] or [basic tutorial][basic-tutorial].
```
Comment (MrMage) [inline DEPRECATION.md]: Also mention (maybe further up) that we consider `nio` production-ready already?

**Comment by MrMage** [comment]
👌

**Comment by MrMage** [comment]
@glbrntt should I merge this?

**Comment by glbrntt** [comment]
It would be useful to have @timburks input here, I think. Let's give him a few more days :)

**Pull request merged**